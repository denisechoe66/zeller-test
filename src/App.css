.selected {
    background-color: #eaf2fa;
    border-radius: 0.5em;
}
