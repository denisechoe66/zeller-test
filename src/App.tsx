import React, { useState, FC } from "react";
import "./App.css";
import User from "./components/User";

const App: FC = () => {
  const [users, setOption] = useState({
    selectedOption: "",
    usersArr: [
      {
        firstName: "John",
        lastName: "Smith",
        role: "Admin",
      },
      {
        firstName: "Adam",
        lastName: "Muller",
        role: "Admin",
      },
      {
        firstName: "Perri",
        lastName: "Smith",
        role: "Manager",
      },
      {
        firstName: "Samsung",
        lastName: "Mobile",
        role: "Manager",
      },
      {
        firstName: "Apple",
        lastName: "iPhone",
        role: "Manager",
      },
    ],
  });

  const onChange = (e: React.FormEvent<HTMLInputElement>) => {
    setOption({ ...users, selectedOption: e.currentTarget.value });
  };

  console.log("options= ", users);

  return (
    <div>
      <h1>User Types</h1>
      <input
        type="radio"
        value="admin"
        name="userType"
        checked={users.selectedOption === "admin"}
        onChange={onChange}
      />{" "}
      Admin
      <input
        type="radio"
        value="manager"
        name="userType"
        checked={users.selectedOption === "manager"}
        onChange={onChange}
      />{" "}
      Manager
      <h1 className="userType">
        {users.selectedOption === "admin"
          ? `${users.selectedOption} Users`
          : users.selectedOption}
      </h1>
      {users.usersArr
        .filter((user) => user.role.toLowerCase() === users.selectedOption)
        .map((user, i) => (
          <div key={i}>
            <User user={user} />
          </div>
        ))}
    </div>
  );
};

export default App;
