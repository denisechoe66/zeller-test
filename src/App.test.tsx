import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import App from './App';

describe('<App />', () => {
  it('should render App', async () => {
    const { container } = render(<App />);
    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });

  it('should see radio buttons change', async () => {
    const { getByLabelText } = render(<App />);
    const radio1 = getByLabelText('Admin') as HTMLInputElement;
    const radio2 = getByLabelText('Manager') as HTMLInputElement;
    fireEvent.click(radio1);
    await waitFor(() => {
      expect(radio1).toBeChecked();
      expect(radio2).not.toBeChecked();
    });
  });

  it('should get mocked data', async () => {
    render(<App />);
    await waitFor(() => {
      expect(screen.queryByText('David Miller')).toBeInTheDocument();
    });
  });
});
