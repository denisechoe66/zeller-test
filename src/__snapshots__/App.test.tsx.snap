// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render App 1`] = `
<div>
  <div>
    <h1
      class="sc-jSgvzq iYBlmY"
    >
      User Types
    </h1>
    <div
      class="sc-gKseQn brWlTu"
    >
      <div
        class="sc-iBPTik VnMyT selected"
      >
        <label
          class="sc-fubCzh fCCgmW"
        >
          <input
            checked=""
            class="sc-pGacB dIScII"
            name="userType"
            type="radio"
            value="admin"
          />
          Admin
        </label>
      </div>
      <div
        class="sc-iBPTik VnMyT"
      >
        <label
          class="sc-fubCzh fCCgmW"
        >
          <input
            class="sc-pGacB dIScII"
            name="userType"
            type="radio"
            value="manager"
          />
          Manager
        </label>
      </div>
    </div>
    <div
      class="sc-gKseQn brWlTu"
    >
      <h1
        class="sc-jSgvzq iYBlmY"
      >
        admin Users
      </h1>
    </div>
  </div>
</div>
`;
